Some Port Channels are not Configured on Leaf Nodes

<!--
If you are reporting a new issue, make sure that we do not have any duplicates
already open. You can ensure this by searching the issue list for this
repository. If there is a duplicate, please close your issue and add a comment
to the existing issue instead.

If you suspect your issue is a bug, please edit your issue description to
include the BUG REPORT INFORMATION shown below. If you fail to provide this
information within 7 days, we cannot debug your issue and will close it. We
will, however, reopen it if you later provide the information.

For more information about reporting issues, see
https://github.com/Azure/SONiC/wiki#report-issues

---------------------------------------------------
GENERAL SUPPORT INFORMATION
---------------------------------------------------

The GitHub issue tracker is for bug reports and feature requests.
General support can be found at the following locations:

- SONiC Support Forums - https://groups.google.com/forum/#!forum/sonicproject

---------------------------------------------------
BUG REPORT INFORMATION
---------------------------------------------------
Use the commands below to provide key information from your environment:
You do NOT have to include this information if this is a FEATURE REQUEST
-->

**Description**

<!--
Briefly describe the problem you are having in a few paragraphs.
-->

**Steps to reproduce the issue:**
1. Use KVM testbed using Sonic Leaf Nodes
2. Run any test (including sanity check) using the `--neighbor_type=sonic` extra parameter

    - i.e. `./run_tests.sh -n vms-kvm-t0 -d vlab-01 -c cacl/test_cacl_function.py -f vtestbed.csv -i veos_vtb -e "--neighbor_type=sonic"`

**Describe the results you received:**

During the sanity check I get the following warning and error:

```
23:09:13 recover.recover                          L0095 WARNING| Try to recover vlab-01 using method adaptive
23:09:13 recover.adaptive_recover                 L0081 WARNING| Restoring {'check_item': 'bgp', 'failed': True, 'host': 'vlab-01', 'bgp': {'down_neighbors': [u'10.0.0.59', u'10.0.0.61', u'10.0.0.63', u'fc00::7e', u'fc00::7a', u'10.0.0.57', u'fc00::76', u'fc00::72']}} with proposed action: config_reload, final action: config_reload
23:09:13 recover.__recover_interfaces             L0036 WARNING| Restoring port: PortChannel0001
23:09:13 recover.__recover_interfaces             L0036 WARNING| Restoring port: PortChannel0003
23:09:13 recover.__recover_interfaces             L0036 WARNING| Restoring port: PortChannel0002
23:09:13 recover.__recover_interfaces             L0036 WARNING| Restoring port: PortChannel0004
23:09:43 recover.adaptive_recover                 L0081 WARNING| Restoring {'check_item': 'interfaces', 'failed': True, 'host': 'vlab-01', 'down_ports': [u'PortChannel0001', u'PortChannel0003', u'PortChannel0002', u'PortChannel0004']} with proposed action: config_reload, final action: config_reload
```

```
E                       Failed: !!!!!!!!!!!!!!!! Pre-test sanity check after recovery failed: !!!!!!!!!!!!!!!!
E                       [
E                           {
E                               "check_item": "bgp",
E                               "failed": true,
E                               "host": "vlab-01",
E                               "bgp": {
E                                   "down_neighbors": [
E                                       "10.0.0.59",
E                                       "10.0.0.61",
E                                       "10.0.0.63",
E                                       "fc00::7e",
E                                       "fc00::7a",
E                                       "10.0.0.57",
E                                       "fc00::76",
E                                       "fc00::72"
E                                   ]
E                               }
E                           },
E                           {
E                               "check_item": "interfaces",
E                               "failed": true,
E                               "host": "vlab-01",
E                               "down_ports": [
E                                   "PortChannel0001",
E                                   "PortChannel0003",
E                                   "PortChannel0002",
E                                   "PortChannel0004"
E                               ]
E                           }
E                       ]
```

In this example, all four portchannels are down. But every time I deploy the minigraph, its a different set of portchannels that are not configured. Below was a case where only portchannels 2 and 3 were configured correctly:

![interfaces_are_weird](https://user-images.githubusercontent.com/31708881/125368685-ac61f100-e32f-11eb-9609-8ab10b8a4e09.png)

On the faulty VMs, the `show interface status` results in the following:

```
admin@ARISTA01T1:~$ show int stat
  Interface    Lanes    Speed    MTU    FEC    Alias    Vlan    Oper    Admin    Type    Asym PFC
-----------  -------  -------  -----  -----  -------  ------  ------  -------  ------  ----------
admin@ARISTA01T1:~$
```


**Describe the results you expected:**
The expected outcome is that all port channels are configured correctly and that the sanity check passes.

On the working VMs, the `show interface status` results in the following:

```
admin@ARISTA02T1:~$ show int stat
   Interface            Lanes    Speed    MTU    FEC           Alias          Vlan    Oper    Admin    Type    Asym PFC
------------  ---------------  -------  -----  -----  --------------  ------------  ------  -------  ------  ----------
   Ethernet0      25,26,27,28      40G   9100    N/A    fortyGigE0/0  PortChannel1      up       up     N/A         N/A
   Ethernet4      29,30,31,32      40G   9100    N/A    fortyGigE0/4        routed      up       up     N/A         N/A
   Ethernet8      33,34,35,36      40G   9100    N/A    fortyGigE0/8        routed      up       up     N/A         N/A
  Ethernet12      37,38,39,40      40G   9100    N/A   fortyGigE0/12        routed      up       up     N/A         N/A
  Ethernet16      45,46,47,48      40G   9100    N/A   fortyGigE0/16        routed    down       up     N/A         N/A
  Ethernet20      41,42,43,44      40G   9100    N/A   fortyGigE0/20        routed    down       up     N/A         N/A
  Ethernet24          1,2,3,4      40G   9100    N/A   fortyGigE0/24        routed    down       up     N/A         N/A
  Ethernet28          5,6,7,8      40G   9100    N/A   fortyGigE0/28        routed    down       up     N/A         N/A
  Ethernet32      13,14,15,16      40G   9100    N/A   fortyGigE0/32        routed    down       up     N/A         N/A
  Ethernet36       9,10,11,12      40G   9100    N/A   fortyGigE0/36        routed    down       up     N/A         N/A
  Ethernet40      17,18,19,20      40G   9100    N/A   fortyGigE0/40        routed    down       up     N/A         N/A
  Ethernet44      21,22,23,24      40G   9100    N/A   fortyGigE0/44        routed    down       up     N/A         N/A
  Ethernet48      53,54,55,56      40G   9100    N/A   fortyGigE0/48        routed    down       up     N/A         N/A
  Ethernet52      49,50,51,52      40G   9100    N/A   fortyGigE0/52        routed    down       up     N/A         N/A
  Ethernet56      57,58,59,60      40G   9100    N/A   fortyGigE0/56        routed    down       up     N/A         N/A
  Ethernet60      61,62,63,64      40G   9100    N/A   fortyGigE0/60        routed    down       up     N/A         N/A
  Ethernet64      69,70,71,72      40G   9100    N/A   fortyGigE0/64        routed    down       up     N/A         N/A
  Ethernet68      65,66,67,68      40G   9100    N/A   fortyGigE0/68        routed    down       up     N/A         N/A
  Ethernet72      73,74,75,76      40G   9100    N/A   fortyGigE0/72        routed    down       up     N/A         N/A
  Ethernet76      77,78,79,80      40G   9100    N/A   fortyGigE0/76        routed    down       up     N/A         N/A
  Ethernet80  109,110,111,112      40G   9100    N/A   fortyGigE0/80        routed    down       up     N/A         N/A
  Ethernet84  105,106,107,108      40G   9100    N/A   fortyGigE0/84        routed    down       up     N/A         N/A
  Ethernet88  113,114,115,116      40G   9100    N/A   fortyGigE0/88        routed    down       up     N/A         N/A
  Ethernet92  117,118,119,120      40G   9100    N/A   fortyGigE0/92        routed    down       up     N/A         N/A
  Ethernet96  125,126,127,128      40G   9100    N/A   fortyGigE0/96        routed    down       up     N/A         N/A
 Ethernet100  121,122,123,124      40G   9100    N/A  fortyGigE0/100        routed    down       up     N/A         N/A
 Ethernet104      81,82,83,84      40G   9100    N/A  fortyGigE0/104        routed    down       up     N/A         N/A
 Ethernet108      85,86,87,88      40G   9100    N/A  fortyGigE0/108        routed    down       up     N/A         N/A
 Ethernet112      93,94,95,96      40G   9100    N/A  fortyGigE0/112        routed    down       up     N/A         N/A
 Ethernet116      89,90,91,92      40G   9100    N/A  fortyGigE0/116        routed    down       up     N/A         N/A
 Ethernet120  101,102,103,104      40G   9100    N/A  fortyGigE0/120        routed    down       up     N/A         N/A
 Ethernet124     97,98,99,100      40G   9100    N/A  fortyGigE0/124        routed    down       up     N/A         N/A
PortChannel1              N/A      40G   9100    N/A             N/A        routed      up       up     N/A         N/A
```

**Additional information you deem important:**
<!--
software/ASIC/Hardware Flatform version and info
-->
**Output of `show version`:**

```
admin@vlab-01:~$ show version
Traceback (most recent call last):
  File "/usr/local/bin/decode-syseeprom", line 18, in <module>
    import sonic_platform
ModuleNotFoundError: No module named 'sonic_platform'

SONiC Software Version: SONiC.master.20135-52c721887
Distribution: Debian 10.9
Kernel: 4.19.0-12-2-amd64
Build commit: 52c721887
Build date: Thu Jun 17 18:40:04 UTC 2021
Built by: AzDevOps@sonic-build-workers-000E7Y

Platform: x86_64-kvm_x86_64-r0
HwSKU: Force10-S6000
ASIC: vs
ASIC Count: 1
Serial Number:
Uptime: 23:41:05 up  6:34,  2 users,  load average: 7.04, 7.63, 8.12

Docker images:
REPOSITORY                    TAG                      IMAGE ID            SIZE
docker-teamd                  latest                   c602171e4172        424MB
docker-teamd                  master.20135-52c721887   c602171e4172        424MB
docker-platform-monitor       latest                   fadbbf955ac9        627MB
docker-platform-monitor       master.20135-52c721887   fadbbf955ac9        627MB
docker-sonic-mgmt-framework   latest                   3ca303078894        570MB
docker-sonic-mgmt-framework   master.20135-52c721887   3ca303078894        570MB
docker-snmp                   latest                   5cd1fb58641a        454MB
docker-snmp                   master.20135-52c721887   5cd1fb58641a        454MB
docker-macsec                 latest                   d55fe0f94fc0        427MB
docker-macsec                 master.20135-52c721887   d55fe0f94fc0        427MB
docker-lldp                   latest                   a2e8c808a629        452MB
docker-lldp                   master.20135-52c721887   a2e8c808a629        452MB
docker-router-advertiser      latest                   c0ad142451a6        412MB
docker-router-advertiser      master.20135-52c721887   c0ad142451a6        412MB
docker-orchagent              latest                   7bb69afa8ce4        442MB
docker-orchagent              master.20135-52c721887   7bb69afa8ce4        442MB
docker-database               latest                   29b0f6494d30        412MB
docker-database               master.20135-52c721887   29b0f6494d30        412MB
docker-nat                    latest                   2db448adbbed        426MB
docker-nat                    master.20135-52c721887   2db448adbbed        426MB
docker-sonic-telemetry        latest                   f41a6396311f        501MB
docker-sonic-telemetry        master.20135-52c721887   f41a6396311f        501MB
docker-dhcp-relay             latest                   2dc6597cc972        419MB
docker-dhcp-relay             master.20135-52c721887   2dc6597cc972        419MB
docker-gbsyncd-vs             latest                   d1d19aaf7333        419MB
docker-gbsyncd-vs             master.20135-52c721887   d1d19aaf7333        419MB
docker-sflow                  latest                   f243b68cd830        424MB
docker-sflow                  master.20135-52c721887   f243b68cd830        424MB
docker-syncd-vs               latest                   39701e06d386        419MB
docker-syncd-vs               master.20135-52c721887   39701e06d386        419MB
docker-fpm-frr                latest                   969442af3204        442MB
docker-fpm-frr                master.20135-52c721887   969442af3204        442MB
```

**Attach debug file `sudo generate_dump`:**

[sonic_dump_vlab-01_20210712_233220.tar.gz](https://github.com/Azure/sonic-mgmt/files/6804940/sonic_dump_vlab-01_20210712_233220.tar.gz)

--- a/ansible/roles/sonic/handlers/main.yml
+++ b/ansible/roles/sonic/handlers/main.yml
@@ -8,7 +8,7 @@
 
 - name: wait for SONiC update config db to finish
   pause:
-    seconds: 60
+    seconds: 180
 
 - name: SONiC restart BGP service
   become: true

--- a/ansible/roles/sonic/handlers/main.yml
+++ b/ansible/roles/sonic/handlers/main.yml
@@ -6,6 +6,10 @@
   become: yes
   listen: "Update config db"
 
+- name: wait for SONiC update config db to finish
+  pause:
+    seconds: 60
+
 - name: SONiC restart BGP service
   become: true
   service: name=bgp

--- a/ansible/roles/sonic/tasks/vsonic.yml
+++ b/ansible/roles/sonic/tasks/vsonic.yml
@@ -48,6 +48,7 @@
   when: hostname in configuration
   notify:
     - Update config db
+    - wait for SONiC update config db to finish
 
 - name: create frr config
   template: src="frr-{{ topo }}-{{ props.swrole }}.j2"

